mdpow/workflows/dihedrals.py

```diff
@@ -633,27 +691,36 @@ def plot_dihedral_violins(df, resname, mol, ab_pairs, figdir=None, molname=None,
         newdir = os.path.join(figdir, subdir)
         os.mkdir(newdir)
 
-    section = df.groupby(by="selection")
+        section = df.groupby(by="selection")
 
-    plot_pdf_width_px = plot_pdf_width * 3.7795275591
+        plot_pdf_width_px = plot_pdf_width * 3.7795275591
 
-    for name in section:
+        for name in section:
         
-        fig = build_svg(mol=mol, molname=molname, name=name, ab_pairs=ab_pairs,
-                        solvents=solvents, width=width)
+            fig = build_svg(mol=mol, molname=molname, atom_group_selection=name, ab_pairs=ab_pairs,
+                            solvents=solvents, width=width)
 
-        figfile = pathlib.Path(newdir) / f"{molname}_{name[0]}_violins.pdf"
-        plot_pdf = cairosvg.svg2pdf(bytestring=fig.tostr(), write_to=str(figfile),
-                                    output_width=plot_pdf_width_px)
+            figfile = pathlib.Path(newdir) / f"{molname}_{name[0]}_violins.pdf"
+            if figdir is not None:
+                plot_pdf = cairosvg.svg2pdf(bytestring=fig.tostr(), write_to=str(figfile),
+                                            output_width=plot_pdf_width_px)
 
-        logger.info(f"Figure saved as {figfile}")
+            logger.info(f"Figure saved as {figfile}")
 
-    logger.info(f"All figures generated and saved in {figdir}")
+        logger.info(f"All figures generated and saved in {figdir}")
 
     return None
 
-def automated_dihedral_analysis(dirname=None, df_save_dir=None, figdir=None,
-                                resname=None, molname=None,
+                                         #figdir=None is a temporary way to satisfy
+                                         #workflows base tests until issue #244 is resolved
+                                         #because it currently uses a **kwargs convention and the
+                                         #positional argument figdir will not carry over nicely, 
+                                         #or requires changes that will result in wasted time
+
+                                         #for this current version, figdir is required and will
+                                         #cause issues if not specified
```

Comment:
indentation is odd — just put all the comments underneath the `def` — or put the `figdir=None` on a line of itself with the comment underneath in the middle of the arg list